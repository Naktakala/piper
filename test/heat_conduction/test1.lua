znodes = {}
N = 10
zmin = 0.0
L = 1.0
znodes = {}

dx = L / N
for i = 0, N do
  znodes[i + 1] = zmin + dx * i
end
meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = { znodes } })
chi_mesh.MeshGenerator.Execute(meshgen1)

system1 = chi_math.KernelSystem.Create
({
  fields = {
    chi_physics.FieldFunctionGridBased.Create({
      name = "T",
      sdm_type = "PWLC"
    })
  },
  kernels = {
    { type = hcm.ThermalConductionKernel.type, var = "T", k = 16.0 }
  },
  bcs = {
    { type = chi_math.FEMDirichletBC.type, var = "T", boundaries = { "ZMAX" }, bc_value = -1.0 },
    { type = chi_math.FEMDirichletBC.type, var = "T", boundaries = { "ZMIN" }, bc_value = 10.0 }
  }
  --verbosity = 2
})

phys1 = chi_math.SteadyNonLinearExecutioner.Create
({
  name = "phys1",
  system = system1,
  solver_params =
  {
    nl_method = "PJFNK",
    l_rel_tol = 1.0e-5,
  }
})

chi.AggregateNodalValuePostProcessor.Create
({
  name = "maxval",
  field_function = "T",
  operation = "max",
  print_on = {"ProgramExecuted"}
})

chi.PostProcessorPrinterSetOptions
(
  {
    print_scalar_time_history = false
  }
)

chiSolverInitialize(phys1)
chiSolverExecute(phys1)

if (master_export == nil) then
  chiExportMultiFieldFunctionToVTK({ "T" }, "test1")
end

chiLogPrintTimingGraph()