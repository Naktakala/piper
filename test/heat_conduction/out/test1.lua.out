mpiexec -np 2 /Users/vermjc/Repos/ChiTech/piper/bin/piper test1.lua --suppress_color --supress_beg_end_timelog master_export=false 
[0]  Parsing argument 1 test1.lua
[0m[0]  Parsing argument 2 --suppress_color
[0m[0]  Parsing argument 3 --supress_beg_end_timelog
[0]  Parsing argument 4 master_export=false
[0]  ChiTech number of arguments supplied: 4
[0]  Computing cell-centroids.
[0]  Done computing cell-centroids.
[0]  Checking cell-center-to-face orientations
[0]  Done checking cell-center-to-face orientations
[0]  00:00:00 Establishing cell connectivity.
[0]  00:00:00 Vertex cell subscriptions complete.
[0]  00:00:00 Surpassing cell 1 of 10 (10%)
[0]  00:00:00 Surpassing cell 2 of 10 (20%)
[0]  00:00:00 Surpassing cell 4 of 10 (30%)
[0]  00:00:00 Surpassing cell 5 of 10 (40%)
[0]  00:00:00 Surpassing cell 6 of 10 (50%)
[0]  00:00:00 Surpassing cell 7 of 10 (60%)
[0]  00:00:00 Surpassing cell 8 of 10 (70%)
[0]  00:00:00 Surpassing cell 9 of 10 (80%)
[0]  00:00:00 Surpassing cell 10 of 10 (90%)
[0]  00:00:00 Establishing cell boundary connectivity.
[0]  00:00:00 Done establishing cell connectivity.
[0]  MeshGenerator: Cells created = 10
[0]  
[0]  Executing solver "phys1"
[0]  NonLinearSolver non-linear iteration    0 - Residual 1.6079801e+03
[0]  NonLinearSolver iteration    0 - Residual 1.6079801e+03
[0]  NonLinearSolver iteration    1 - Residual 1.8229008e-06
[0]  NonLinearSolver non-linear iteration    1 - Residual 7.2095060e-05
[0]  NonLinearSolver iteration    0 - Residual 7.2095060e-05
[0]  NonLinearSolver iteration    1 - Residual 5.0348276e-12
[0]  NonLinearSolver non-linear iteration    2 - Residual 2.9804875e-12
[0]            Converged CONVERGED_FNORM_ABS
[0]  
[0]  
[0]  SCALAR post-processors latest values at event "ProgramExecuted"
[0]  *----------------------*-----------------*
[0]  | Post-Processor Name  | Value           |
[0]  *----------------------*-----------------*
[0]  | maxval(latest)       |       10.000000 |
[0]  *----------------------*-----------------*
[0]  


